contracts/src/v0.8/dev/KeeperRegistryBase.sol

```diff
@@ -230,12 +261,21 @@ abstract contract KeeperRegistryBase is ConfirmedOwner, ExecutionPrevention, Ree
     address from,
     uint256 id,
     bytes memory performData,
-    bool useTxGasPrice
+    bool isExecution
   ) internal view returns (PerformParams memory) {
     uint256 gasLimit = s_upkeep[id].executeGas;
     (uint256 gasWei, uint256 linkEth) = _getFeedData();
-    uint256 adjustedGasWei = _adjustGasPrice(gasWei, useTxGasPrice);
-    uint96 maxLinkPayment = _calculatePaymentAmount(gasLimit, adjustedGasWei, linkEth);
+    uint256 adjustedGasWei = _adjustGasPrice(gasWei, isExecution);
+    bytes memory data = new bytes(0);
+    if (PAYMENT_MODEL == PaymentModel.OPTIMISM) {
+      if (isExecution) {
+        data = bytes.concat(msg.data, L1_FEE_DATA_PADDING);
+      } else {
+        data = bytes.concat(performData, PERFORM_DATA_PADDING);
```

Comment:
Actually thinking more we need more than just `PERFORM_DATA_PADDING`. This needs to include function selector for perform, id of upkeep and length of performData so 4+32+32 bytes padding

Reply:
and you can also include `L1_FEE_DATA_PADDING` inside `PERFORM_DATA_PADDING` to prevent separate line below

Reply:
should it be 4 bytes for function selector + 32 bytes for upkeep id + performData + data padding (17 bytes)? at this function we already know the perform data size so we don't need adding another 32 bytes?

Reply:
What I have seen is another 32 bytes included to indicate the length of performData followed by performData itself. e.g. See the calldata for empty performData (0x): https://optimistic.etherscan.io/tx/0x3ea6c4b89e40ecd789f34196ac46093e3be29a524e9db0290f3d07477710838f

Reply:
Done